internal/consensus/state.go

```diff
@@ -541,7 +530,9 @@ func (cs *State) OnStop() {
 		select {
 		case <-cs.getOnStopCh():
 		case <-time.After(cs.state.ConsensusParams.Timeout.Commit):
+			cs.mtx.RLock()
```

Comment:
do we need locking just for logs? I wonder if we can ignore race cond for this. I'm worried that this could expose us to a deadlock if something else needs cs.mtx while calling OnStop()

Reply:
good point. I'll remove it

Reply:
actually it seems like removing it would very often cause the unit test to fail. I'll add it back for now